### PR TITLE
ROX-9539: ScopeCheckerCore for tests

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

New object to inject Scopes for access control checks in tests

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

New unit tests
Replace OneStepSCC in the codebase by the new object and ensure unit tests still work

### CODE REVIEW

File: pkg/sac/test_scope_checker_core.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+			core.scope[access] = make(map[permissions.Resource]*TestResourceScope, 0)
+		}
+		core.scope[access][resource.Resource.GetResource()] = includedResourceScope
+	}
+	return core
+}
+
+// TestScopeCheckerCoreFromFullScopeMap creates a ScopeCheckerCore that allows scoped access to the input
+// scope tree for testing purposes.
+func TestScopeCheckerCoreFromFullScopeMap(_ *testing.T, targetScope map[storage.Access]map[permissions.Resource]*TestResourceScope) ScopeCheckerCore {
```
Comment (janisz) [inline pkg/sac/test_scope_checker_core.go:52]: How about creating an alias for `map[storage.Access]map[permissions.Resource]*TestResourceScope` to make usage simpler? It's easier to type&read `sac.ScopeMap`

File: pkg/sac/test_scope_checker_core.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		scope: targetScope,
+	}
+}
+
+func (c *testScopeCheckerCore) EffectiveAccessScope(resource permissions.ResourceWithAccess) (*effectiveaccessscope.ScopeTree, error) {
+	resourceMap := c.scope[resource.Access]
+	if len(resourceMap) == 0 {
+		return effectiveaccessscope.DenyAllEffectiveAccessScope(), nil
+	}
+	resourceCore := resourceMap[resource.Resource.GetResource()]
```
Comment (md2119) [inline pkg/sac/test_scope_checker_core.go:66]: `resource.Resource.GetResource()` reads funny

File: pkg/sac/test_scope_checker_core.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+)
+
+// TestClusterScope contains cluster-level scope information (cluster included or list of included namespaces
+// in the cluster) to build test scope checkers.
+type TestClusterScope struct {
+	Namespaces []string
+	Included   bool
+}
+
+// TestResourceScope contains resource-level scope information (resource fully included or list of included clusters
```
Comment (md2119) [inline pkg/sac/test_scope_checker_core.go:19]: by `resource fully included` do you mean global access?

File: pkg/sac/test_scope_checker_core.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	resourceMap := c.scope[resource.Access]
+	if len(resourceMap) == 0 {
+		return effectiveaccessscope.DenyAllEffectiveAccessScope(), nil
+	}
+	resourceCore := resourceMap[resource.Resource.GetResource()]
+	if resourceCore == nil || !resourceCore.Included || len(resourceCore.Clusters) == 0 {
+		return effectiveaccessscope.DenyAllEffectiveAccessScope(), nil
+	}
+	if resourceCore.Included {
+		return effectiveaccessscope.UnrestrictedEffectiveAccessScope(), nil
```
Comment (md2119) [inline pkg/sac/test_scope_checker_core.go:71]: Is `resourceCore.Included` supposed to mean global access?

File: pkg/sac/test_scope_checker_core.go (hunk 3 above)
Comment (rhybrillou) [reply pkg/sac/test_scope_checker_core.go:19]: Indeed, resource fully included means global access (that is, access to objects of the resource type in any cluster and namespace).

**Pull request merged**